experimental/PropertyDDS/packages/property-shared-tree-interop/src/schemaConverter.ts

```diff
@@ -368,6 +378,47 @@ function enhanceRootFieldSchemaWithChildren(
 	return enhancedRootFieldSchema;
 }
 
+/**
```

Comment:
@milanro I have added a helper function to dynamically create types for arrays, maps and sets. However, I still think it's better to have pre-build types for primitives, as it only works via shallow copy, which could be costly.

Reply:
@sharptrip I guess that it is fine to have predefined arrays and maps of primitives. Please, go on with merge.